pkg/git/git.go

```diff
@@ -207,57 +232,67 @@ func (r *Repo) RevParseShort(rev string) (string, error) {
 
 // LatestNonPatchFinalToLatest tries to discover the start (latest v1.xx.0) and
 // end (release-1.xx or master) revision inside the repository
-func (r *Repo) LatestNonPatchFinalToLatest() (start, end string, err error) {
+func (r *Repo) LatestNonPatchFinalToLatest() (*DiscoverResult, error) {
 	// Find the last non patch version tag, then resolve its revision
 	versions, err := r.latestNonPatchFinalVersions()
 	if err != nil {
-		return "", "", err
+		return nil, err
 	}
 	version := versions[0]
 	versionTag := addTagPrefix(version.String())
-	logrus.Infof("latest non patch version %s", versionTag)
-	start, err = r.RevParse(versionTag)
+	logrus.Debugf("latest non patch version %s", versionTag)
+	start, err := r.RevParse(versionTag)
 	if err != nil {
-		return "", "", err
+		return nil, err
 	}
 
 	// If a release branch exists for the next version, we use it. Otherwise we
 	// fallback to the master branch.
-	end, err = r.releaseBranchOrMasterRev(version.Major, version.Minor+1)
+	end, branch, err := r.releaseBranchOrMasterRev(version.Major, version.Minor+1)
 	if err != nil {
-		return "", "", err
+		return nil, err
 	}
 
-	return start, end, nil
+	return &DiscoverResult{
```

Comment:
For me personally it makes error handling easier because I always expect that the returned pointer is not nil if `err == nil`. This way we could also return `nil` in case of an error. If we stick to the struct without the pointer, I would have to return `DiscoverResult{}, err...`, which bloats up the verbosity of the source code. WDYT? :)

Reply:
This is obviously a bit personal preference, but some things I try to do:
- Make zero-values work & useful
- Prefer value receivers if I don't need to e.g. manipulate the receiver

I def won't block on that, just suggestions ...

Reply:
Okay, change it to `DiscoverResult{}` :+1: